### DISCUSSION TITLE
SharePoint online PnP command to find users and delete it if exists

### DISCUSSION THREAD

**Discussion by santoshsubedi10** [pnp/powershell#3953] [Q&A]
I am trying to create a script where it runs on site collection to find certain users. (ex 'Everyone' and 'everyone except external').
If those users doesn't exist, do nothing but if those exist, remove the user.

I figured how to remove a user. But how do i scan the users and then remove it all at once ? 

$SiteURL = "https://test"
$UserID= ("everyone")
  
Connect-PnPOnline -Url $SiteUrl -UseWebLogin

$User = Get-PnPUser -Identity $UserID 
  
if ($User) 
   {
   Remove-PnPUser -Identity $UserID
   Write-host -f Green "Removed User $UserID from the site!"
 } 
Else
{
    Write-host -f Yellow "User $UserID doesn't exist on the site!"

**Comment by Barbarur** [comment]
The command Remove-PnPUser checks for the user in the site and then removes it. If the user is not on the site, it will throw an error. 

The groups "Everyone" and "Everyone except external users" are registered in all sites by default, even if they don't have any permissions. You will find it in all the sites.

The $UserID for Everyone is wrong. You can check the correct LoginName easily from SharePoint. I think it's "c:0(.s|true".
